grails-app/services/org/pih/warehouse/core/DocumentService.groovy

```diff
@@ -934,6 +931,43 @@ class DocumentService {
         }
     }
 
+    void generateInventoryTemplate(OutputStream outputStream, List<Map> data) {
+        try {
+            Workbook workbook = new HSSFWorkbook()
+            CreationHelper createHelper = workbook.getCreationHelper()
+            HSSFSheet sheet = workbook.createSheet("Sheet1")
+            createExcelHeader(sheet, 0, data.get(0).keySet().toList())
+            data.eachWithIndex { Map dataRow, rowIndex ->
+                Row excelRow = sheet.createRow(rowIndex + 1)
+                dataRow.keySet().eachWithIndex { columnName, cellIndex ->
+                    def cellValue = dataRow.get(columnName) ?: ""
+                    // POI can't handle objects so we need to convert all objects to strings unless they are numeric
+                    if (!(cellValue instanceof Number)) {
+                        cellValue = cellValue.toString()
+                    }
+                    excelRow.createCell(cellIndex)
+                    if (columnName == 'Expiration date' && cellValue != '') {
```

Comment:
For example, instead of hardcoding this here, we could change the row data to include a dataType (date) and format ("MM/dd/yyyy"). For dataType = Number we toString() the cellValue.